lib/libnv/msgio.c

```diff
@@ -63,13 +63,21 @@ __FBSDID("$FreeBSD$");
 /* Linux: arbitrary size, but must be lower than SCM_MAX_FD. */
 #define	PKG_MAX_SIZE	((64U - 1) * CMSG_SPACE(sizeof(int)))
 #else
+#ifndef __CHERI_PURE_CAPABILITY__
 /*
  * To work around limitations in 32-bit emulation on 64-bit kernels, use a
  * machine-independent limit on the number of FDs per message.  Each control
  * message contains 1 FD and requires 12 bytes for the header, 4 pad bytes,
  * 4 bytes for the descriptor, and another 4 pad bytes.
  */
 #define	PKG_MAX_SIZE	(MCLBYTES / 24)
+#else
+/*
+ * For CheriABI we have the descriptor by 12 bytes of pad for capability
```

Comment:
s/by/be/?

FWIW, I usually prefer using #ifdef instead of the negative logic, but I can appreciate that you probably want the purecap comment to read as a continuation of the previous comment?

Reply:
Reworked to have a single block comment and positive logic.